### PR TITLE
Move DFHack version to a separate (static) library

### PR DESCRIPTION
**Pull request by lethosor**
This allows the DFHack version to be changed without recompiling everything - only DFHackVersion.cpp (a two-line file) needs to be recompiled.
(Any structure/include changes will still cause source files that depend on them to be rebuilt.)

### CODE REVIEW

**Comment by expwnent** [comment]
Will make/cmake correctly detect that it needs to update DFHackVersion even though the file itself hasn't changed when the version macro has?

**Comment by lethosor** [comment]
Yes. Since DFHACK_VERSION is a preprocessor definition, CMake will rebuild any files that it is defined for when it changes. (This is why changing it previously required rebuilding _every_ file, since DFHACK_VERSION was a project-wide definition.)

File: library/CMakeLists.txt [hunk 1]
```diff
@@ -249,6 +250,9 @@ ELSE(WIN32)
   SET(PROJECT_LIBS psapi dfhack-tinyxml dfhack-tinythread)
 ENDIF()
 
+ADD_LIBRARY(dfhack-version STATIC DFHackVersion.cpp)
+SET_TARGET_PROPERTIES(dfhack-version PROPERTIES COMPILE_DEFINITIONS DFHACK_VERSION="${DFHACK_VERSION}")
+
```
Comment (lethosor) [inline library/CMakeLists.txt]: This should work on Windows, but it would be nice if somebody could test it before merging.

**Comment by expwnent** [comment]
Seems to work.

**Pull request merged**